internal/api/provider/azure.go

```diff
@@ -34,15 +38,19 @@ func NewAzureProvider(ext conf.OAuthProviderConfiguration, scopes string) (OAuth
 		return nil, err
 	}
 
-	authHost := chooseHost(ext.URL, defaultAzureAuthBase)
-	apiPath := chooseHost(ext.ApiURL, defaultAzureAPIBase)
-
 	oauthScopes := []string{"openid"}
 
 	if scopes != "" {
 		oauthScopes = append(oauthScopes, strings.Split(scopes, ",")...)
 	}
 
+	authHost := chooseHost(ext.URL, defaultAzureAuthBase)
+	expectedIssuer := ""
+
+	if ext.URL != "" {
+		expectedIssuer = authHost + "/v2.0"
+	}
```

Comment:
Yeah, because the issuer of the ID token is not the authorize URL, but it is if you call the tenant's authorize. So basically, if you use common then any issuer is acceptable, but if you use a tenant's authorize then only that tenant's issuer is acceptable.